### PR TITLE
consul_kv read configuration from consul 

### PR DESCRIPTION
**Pull request by raben2**
##### 

```
Feature Pull Request
```

COMPONENT NAME

clustering/consul_kv.py
ANSIBLE VERSION

ansible 2.2.0
config file = /etc/ansible/ansible.cfg
configured module search path = ['/home/work/scripts/ansible-modules/*']
SUMMARY

added module state 'read' to obtain information from consul key value store
example:

```
name: Read from consul key consul_kv key: test/key1 state: read
```

output:

ok: [localhost] => {
"msg": {
"changed": true,
"data": {
"CreateIndex": 370684,
"Flags": 0,
"Key": "test/key1",
"LockIndex": 0,
"ModifyIndex": 379139,
"Value": "SomeValueFromKeystore "
},
"index": "379139",
"key": "FFB/stage/versions"
}
}

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @raben2. To the current maintainers, @sgargan please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: clustering/consul_kv.py [hunk 1]
```diff
@@ -158,9 +162,24 @@ def execute(module):
...
     if state == 'present':
         add_value(module)
+    if state == 'read':
+        read_key(module, state)
     else:
         remove_value(module)
 
+def read_key(module, state):
+  consul_api = get_consul_api(module)
+  session = module.params.get('session')
```
Comment (sgargan) [inline clustering/consul_kv.py:172]: ⚡ the session as its not used.

File: clustering/consul_kv.py [hunk 2]
```diff
@@ -158,9 +162,24 @@ def execute(module):
         lock(module, state)
     if state == 'present':
         add_value(module)
+    if state == 'read':
+        read_key(module, state)
     else:
         remove_value(module)
 
+def read_key(module, state):
+  consul_api = get_consul_api(module)
```
Comment (sgargan) [inline clustering/consul_kv.py:171]: indent 4 to be consistent.

File: clustering/consul_kv.py [hunk 3]
```diff
@@ -158,9 +162,24 @@ def execute(module):
...
+  session = module.params.get('session')
+  key = module.params.get('key')
+  try: 
+    index, value = consul_api.kv.get(key)
+  except Exception, e:
+     module.fail_json(msg="something went wrong: %s" % e)
```
Comment (sgargan) [inline clustering/consul_kv.py:177]: Would be good to have some context in the error message

``` python
module.fail_json(msg="Error reading '%s' key from consul: %s" % (key, e))
```

File: clustering/consul_kv.py [hunk 4]
```diff
@@ -158,9 +162,24 @@ def execute(module):
...
+  changed = True         
+  module.exit_json(changed=changed,
```
Comment (sgargan) [inline clustering/consul_kv.py:179]: just inline the change here

**Review by sgargan — CHANGES_REQUESTED** [review]
This is great @raben2 thanks for taking the time to add it. I've a couple of very minor comments. 

If you're feeling diligent, there are tests in the core project (here)[https://github.com/ansible/ansible/blob/devel/test/integration/roles/test_consul_kv/tasks/main.yml] would be awesome to get one for your read function in there too.

File: clustering/consul_kv.py (hunk 4 above)
Comment (raben2) [reply clustering/consul_kv.py:179]: thank you for the additions!

**Comment by gregdek** [comment]
Thanks @raben2 for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@raben2 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**